testutils/util/artifact.py

```diff
@@ -55,7 +75,21 @@ def __init__(
         if payload is not None:
             self.add_payload(payload, payload_type, depends, provides)
 
-    def add_payload(self, fd, payload_type="rootfs_image", depends=None, provides=None):
+    def add_state_script(self, state, script):
```

Comment:
This is fine, but unrelated to the PR. Why?

Reply:
I was thinking of making a state-script no-op artifact, but figured out an easier way once I had already implemented it.